File: src/test/java/utlities/WordCountTest.java
package utlities;

import org.junit.jupiter.api.Test;
import utilities.WordCounter;

import static org.junit.jupiter.api.Assertions.*;


public class WordCountTest {

    @Test
    public void testCountWordsWithAllValidWords(){

        String input = "Mary had a little lamb";

        int countedWords = WordCounter.countWords(input);

        assertEquals(5, countedWords);
    }

    @Test
    public void testCountWordsWithTrailingSpacesAndInvalidWords(){

        String input = "    Mary h3d a little   lamb   ";

        int countedWords = WordCounter.countWords(input);

        assertEquals(4, countedWords);
    }

    @Test
    public void testCountWordsWithNullInput(){

        String input = null;

        int countedWords = WordCounter.countWords(input);

        assertEquals(0, countedWords);
    }

    @Test
    public void testIsValidWordForValidWord(){

        String input = "Mary";

        boolean isValidWord = WordCounter.isValidWord(input);

        assertTrue(isValidWord);
    }

    @Test
    public void testIsValidWordForInvalidWordWithNumbers(){

        String input = "H3llo";

        boolean isValidWord = WordCounter.isValidWord(input);

        assertFalse(isValidWord);
    }

    @Test
    public void testIsValidWordForInvalidWordWithSpecialCharacters(){

        String input = "H ll$o";

        boolean isValidWord = WordCounter.isValidWord(input);

        assertFalse(isValidWord);
    }

}

File: src/test/kotlin/at/erste/WordCountTest.kt
package at.erste

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class WordCountTest {

    @Test
    fun testMultiplePunctuation() {
        val input = "Hello, world!!!"
        val expected = 2
        val actual = countWords(input)
        assertEquals(expected, actual)
    }

    @Test
    fun testSpecialCharacter() {
        val input = "Hello, my name is Stan9slav!!! How are you?"
        val expected = 7
        val actual = countWords(input)
        assertEquals(expected, actual)
    }

    @Test
    fun testEmptyInput() {
        val input = ""
        val expected = 0
        val actual = countWords(input)
        assertEquals(expected, actual)
    }
}
